arrows/mvg/camera_options.cxx

```diff
@@ -0,0 +1,206 @@
+  template <>                                                             \
+  std::string                                                             \
+  mvg_options< NS::ceres_type >()                                         \
+  {                                                                       \
+    typedef NS::ceres_type T;                                             \
+    std::string options_str = "\nMust be one of the following options:";  \
+    std::string opt;                                                      \
+    for( unsigned i = 0; i < 20; ++i )                                    \
+    {                                                                     \
+      opt = NS::ceres_type##ToString( static_cast< T >( i ) );            \
+      if( opt == "UNKNOWN" )                                              \
+      {                                                                   \
+        break;                                                            \
+      }                                                                   \
+      options_str += "\n  - " + opt;                                      \
+    }                                                                     \
+    return options_str;                                                   \
+  }                                                                       \
+                                                                          \
+  }                                                                       \
+  }                                                                       \
+  }
+
+MVG_ENUM_HELPERS( kwiver::arrows::mvg, LensDistortionType )
+
+#undef MVG_ENUM_HELPERS
+
+namespace kwiver {
+
+using namespace vital;
+
+namespace arrows {
+
+namespace mvg {
+
+#define CASESTR( x ) case x: return #x
+#define STRENUM( x ) if( value == #x ) { *type = x; return true; }
+
+/// Convert a string to upper case.
+static void
+UpperCase( std::string* input )
+{
+  std::transform( input->begin(), input->end(), input->begin(), ::toupper );
+}
+
+/// Provide a string representation for a LensDisortionType value.
+const char*
+LensDistortionTypeToString( LensDistortionType type )
+{
+  switch( type )
+  {
+    CASESTR( NO_DISTORTION );
+    CASESTR( POLYNOMIAL_RADIAL_DISTORTION );
+    CASESTR( POLYNOMIAL_RADIAL_TANGENTIAL_DISTORTION );
+    CASESTR( RATIONAL_RADIAL_TANGENTIAL_DISTORTION );
+    default:
+      return "UNKNOWN";
+  }
+}
+
+/// Parse a LensDistortionType value from a string or return false.
+bool
+StringToLensDistortionType( std::string value, LensDistortionType* type )
+{
+  UpperCase( &value );
+  STRENUM( NO_DISTORTION );
+  STRENUM( POLYNOMIAL_RADIAL_DISTORTION );
+  STRENUM( POLYNOMIAL_RADIAL_TANGENTIAL_DISTORTION );
+  STRENUM( RATIONAL_RADIAL_TANGENTIAL_DISTORTION );
+  return false;
+}
+
+camera_options
+::camera_options()
+  : optimize_focal_length( true ),
```

Comment:
Inline member initializers might be nice?

Reply:
inline init